### PR TITLE
Feature/plugins

### PR DESCRIPTION
**Pull request by dbmeneses**
ITs or medium tests still missing

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+import javax.annotation.Nullable;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class AdapterModuleFileSystem implements ModuleFileSystem {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java]: MINOR Remove this use of "ModuleFileSystem"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  @Override
+  public List<File> files(FileQuery query) {
+    Collection<FilePredicate> predicates = Lists.newArrayList();
+    for (Map.Entry<String, Collection<String>> entry : query.attributes().entrySet()) {
+      predicates.add(fromDeprecatedAttribute(entry.getKey(), entry.getValue()));
+    }
+    return ImmutableList.copyOf(sonarLintFileSystem.files(predicates().and(predicates)));
+  }
+
+  private FilePredicate fromDeprecatedAttribute(String key, Collection<String> value) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java]: MAJOR The Cyclomatic Complexity of this method "fromDeprecatedAttribute" is 18 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<File> binaryDirs() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public List<File> files(FileQuery query) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/analysis/filesystem/AdapterModuleFileSystem.java]: MINOR Remove this use of "FileQuery"; it is deprecated. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/GlobalPropertiesDownloader.java [hunk 4]
```diff
@@ -81,7 +106,18 @@ private static void parseProperty(JsonReader reader, GlobalProperties.Builder bu
       String[] list = globalProperties.getProperties().get("sonarlint.plugins.whitelist").split(",");
       return ImmutableSet.copyOf(list);
     }
-    return ImmutableSet.of("java", "javascript", "php", "python");
+    return ImmutableSet.of("java", "javascript", "php", "python", "cobol", "abap", "plsql", "cpp");
+  }
+
+  private boolean needsBatchGlobal(String version) {
```
Comment (SonarTech) [inline core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/GlobalPropertiesDownloader.java]: MAJOR Make "needsBatchGlobal" a "static" method. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
Let's wait to see if ITs are green but LGTM

**Pull request merged**